test/junit/scala/tools/nsc/parser/ParserTest.scala

```diff
@@ -0,0 +1,32 @@
+package scala.tools.nsc.parser
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.tools.testing.BytecodeTesting
+
+@RunWith(classOf[JUnit4])
+class ParserTest extends BytecodeTesting{
+  override def compilerArgs: String = "-Ystop-after:parser -Yvalidate-pos:parser -Yrangepos"
+  @Test
+  def crlfRangePositionXml_t10321(): Unit = {
+    val code =
+      """
+        |object Test {
+        |  Nil.map { _ =>
+        |    <X />
+        |    <Y />
+        |  }
+        |}
+      """.stripMargin
+    val crlfCode = code.replaceAllLiterally("\n", "\r\n")
+    val lfCode = code
+    assert(crlfCode != lfCode)
+    import compiler._, global._
+    val run = new Run
+    run.compileSources(newSourceFile(lfCode) :: Nil)
+    assert(!reporter.hasErrors)
+    run.compileSources(newSourceFile(crlfCode) :: Nil)
```

Comment:
surely this is wrong. It depends of the current line endings, and you are assuming that the file has unix line endings?

```
//to cope with the case of windows or unix line endings, start with unix style 
val strippedCode = code.replaceAllLiterally("\r\n", "\n") 
val crlfCode = strippedCode.replaceAllLiterally("\n", "\r\n") 
val lfCode = strippedCode 
```

Reply:
see the second commit and comment on other PR.

Reply:
Ah - RACE CONDITION - 
you fixed it as I was writing the comment
😆

Reply:
Also squashed, sorry about that, Jenkins.